frontend/components/forms/image-uploader/component.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { ChangeEvent, useState } from 'react';
+
+import { FieldValues } from 'react-hook-form';
+import { useIntl } from 'react-intl';
+
+import Image from 'next/image';
+
+import Button from 'components/button';
+import Loading from 'components/loading';
+
+import { ImageUploaderProps } from './types';
+
+export const ImageUploader = <FormValues extends FieldValues>({
+  preview,
+  register,
+  name,
+  id,
+  text,
+  handleChangeImage,
+  registerOptions,
+  ...rest
+}: ImageUploaderProps<FormValues>) => {
+  const { formatMessage } = useIntl();
+  const [imagePreview, setImagePreview] = useState<string>();
+
+  const handleUploadImage = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.currentTarget.files?.length) {
+      const file = e.currentTarget.files[0];
+      const src = URL.createObjectURL(file);
+      setImagePreview(src);
+      handleChangeImage();
+    }
+  };
+
+  return (
+    <div className="flex items-center justify-start">
+      {preview && (
+        <div className="mr-5">
+          <Image
+            src={imagePreview || '/images/avatar.svg'}
+            width={48}
+            height={48}
+            className="rounded-full"
+            alt={formatMessage({ defaultMessage: 'profile image', id: 'OX08fr' })}
+          />
+        </div>
+      )}
+      <Button>
+        <Loading className="inline mr-4" />
+        {text || formatMessage({ defaultMessage: 'Upload Image', id: 'MntrZe' })}
+        <input
```

Comment:
🔴

You cannot put an input in a button. That's invalid HTML. Instead, use the native button of the file input. You can see how to style it in [Tailwind's documentation](https://tailwindcss.com/docs/hover-focus-and-other-states#file-input-buttons).

The UI will be slightly different from the design (because the text will be “Browser…” and the image's path will be displayed) but that avoids having to implement a more complex custom component.